src/lib/scene-internal.ts

```diff
@@ -721,6 +721,10 @@ export class SceneInternal implements Renderer {
     const {width, height} = this.canvas.getBoundingClientRect();
 
     if (!width || !height) {
+      console.warn('Delaying Scene initialization: canvas has zero size');
```

Comment:
There are two reasons why an element may have zero height or width. One as that those properties may be genuinely zero. The other is that the element is inheriting its size from a parent node, but somewhere up the chain it is not attached to the DOM and therefore has no computable size.

If the canvas has no width or height (for either reason) then it's inappropriate to compute initialization parameters. This could be the API user's intent, but more often than not it indicates that the API user (or, more specifically, the framework that the API user is working within) has not yet attached the container to the DOM. So here we report the situation to the console so that the API user can diagnose the problem.